tools/ascli-actions-mdata.c

```diff
@@ -252,6 +252,50 @@ ascli_what_provides (const gchar *cachepath, const gchar *kind_str, const gchar
 	return 0;
 }
 
+/**
+ * ascli_by_categories:
+ *
+ * Get components that match the given @p categories
+ */
+int
+ascli_by_categories (const gchar *cachepath, const gchar *categories, gboolean detailed, gboolean no_cache)
+{
+    g_autoptr(AsPool) dpool = NULL;
+    g_autoptr(GPtrArray) result = NULL;
+    g_autoptr(GError) error = NULL;
+
+    if (categories == NULL) {
+        fprintf (stderr, "%s\n", _("You need to specify a categories to look up."));
+        return 2;
+    }
+
+    dpool = ascli_data_pool_new_and_open (cachepath, no_cache, &error);
+    if (error != NULL) {
+        g_printerr ("%s\n", error->message);
+        return 1;
+    }
+
+    g_auto(GStrv) cats = g_strsplit(categories, ",", 0);
```

Comment:
No declarations after statements please, we (still) need to be compatible with older C versions/compilers.

Reply:
I don't think I've ever seen a compiler that _required_ it and I've been doing C for about 20 years now.